librz/core/rop.c

```diff
@@ -416,29 +444,62 @@ RZ_API bool rz_core_rop_gadget_info_has_register(const RZ_NONNULL RzRopGadgetInf
 	return false;
 }
 
-static bool is_var_write_event(const RzRopRegInfo *reg_info) {
+static inline bool is_var_read_event(const RzRopRegInfo *reg_info) {
+	return reg_info->is_var_read && !reg_info->is_mem_read && !reg_info->is_var_write && !reg_info->is_mem_write;
+}
+
+static inline bool is_var_write_event(const RzRopRegInfo *reg_info) {
 	return reg_info->is_var_write;
 }
 
-static bool is_mem_read_event(const RzRopRegInfo *reg_info) {
+static inline bool is_mem_read_event(const RzRopRegInfo *reg_info) {
 	return reg_info->is_mem_read;
 }
 
-static bool is_mem_write_event(const RzRopRegInfo *reg_info) {
+static inline bool is_mem_write_event(const RzRopRegInfo *reg_info) {
 	return reg_info->is_mem_write;
 }
 
-static bool is_pc_write_event(const RzRopRegInfo *reg_info) {
+static inline bool is_pc_write_event(const RzRopRegInfo *reg_info) {
 	return reg_info->is_pc_write;
 }
 
 rz_rop_event_check_fn rz_rop_event_functions[RZ_ROP_EVENT_COUNT] = {
+	is_var_read_event,
 	is_var_write_event,
 	is_mem_read_event,
 	is_mem_write_event,
 	is_pc_write_event,
 };
 
+/**
+ * \brief Check if a given event dependency is present for a register
+ * \param gadget_info RZ_NONNULL Pointer to the RzRopGadgetInfo object.
+ * \param event The RzRopEvent to check.
+ * \param reg_name Name of the register
+ * \return True if there is a match or false
```

Comment:
```suggestion
 * \return True if there is an event which \p event which uses \p reg_name. False otherwise.
```

Also as above. Please give a description to the parameters, which describes the purpose they have in the function.

e.g. `\param reg_name The register name to filter the events.`

Reply:
Done :+1: